homeassistant/components/color_extractor/__init__.py

```diff
@@ -0,0 +1,171 @@
+"""Module for color_extractor (RGB extraction from images) component."""
+import asyncio
+import io
+import logging
+
+from PIL import UnidentifiedImageError
+import aiohttp
+import async_timeout
+from colorthief import ColorThief
+import voluptuous as vol
+
+from homeassistant.components.color_extractor.const import (
+    ATTR_FILE_PATH,
+    ATTR_URL,
+    DOMAIN,
+    SERVICE_PREDOMINANT_COLOR_FILE,
+    SERVICE_PREDOMINANT_COLOR_URL,
+)
+from homeassistant.components.light import (
+    ATTR_RGB_COLOR,
+    DOMAIN as LIGHT_DOMAIN,
+    LIGHT_TURN_ON_SCHEMA,
+    SERVICE_TURN_ON,
+)
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.helpers import aiohttp_client
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+# Extend the existing light.turn_on service schema
+# with the addition of our ATTR_FILE_PATH
+FILE_SERVICE_SCHEMA = cv.make_entity_service_schema(
+    {
+        **LIGHT_TURN_ON_SCHEMA,
+        vol.Required(ATTR_ENTITY_ID): cv.string,
+        vol.Required(ATTR_FILE_PATH): cv.string,
+    }
+)
+
+URL_SERVICE_SCHEMA = cv.make_entity_service_schema(
```

Comment:
I think that it would make sense to merge these two schemas, and use 

```python
vol.Exclusive('extract_path', 'color_extractor'): cv.path,
vol.Exclusive('extract_url', 'color_extractor'): cv.url,
```

that way only 1 is allowed to be specified. By prefixing it with `extract_`, we can be more sure that it will never overlap with light service params.